### ISSUE TITLE
Rebooting takes place without the user having the chance to remove his/her medium - Fix from ROSA inside (Bugzilla Bug 276)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#276]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 276
Date: 2013-11-06 22:37:53 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;
To:   OpenMandriva QA &lt;<bugs@openmandriva.org>&gt;
CC:   Wayne@WayneSallee.com, @benbullard79, @cris-b, denis.silakov@rosalab.ru, @itchka, @tapwag, @tpgxyz

See also:     https://issues.openmandriva.org/show_bug.cgi?id=1446
See also:     https://issues.openmandriva.org/show_bug.cgi?id=1079
Last updated: 2019-02-06 20:43:12 +0000

## Comment 1942
Date: 2013-11-06 22:37:53 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

At the end of the first part of installation process installation media (usb or dvd) must be removed. This is not an easy task for beginner because you must remove media when pc is off and before reboot begin. But reboot start almost immediately and if you don't remove media installation restart!

I think you should complete first part of installation process and at the very end add a “pause” before shutoff, issue a message asking for removing media. And before complete shut off waiting for an user action like pressing a button. This give normal user enough time to remove media safely.

## Comment 1943
Date: 2013-11-06 23:25:21 +0000
From: @benbullard79

Makes sense to me. One of several details that would make live installer more professional/polished.

## Comment 1949
Date: 2013-11-07 06:28:07 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

I forgot a feature: when using dvd automatically open dvd reader when media has to be removed.

## Comment 3110
Date: 2014-02-04 16:07:19 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Still valid in 2014 alpha1 64 bit.
Please add a "pause" after completing the first part of installation, just before reboot.
It should not so difficult.

## Comment 3806
Date: 2014-03-12 11:57:15 +0000
From: @itchka

Upping status of this bug should implemented before final release of 2014 if it's not a huge task.

## Comment 4453
Date: 2014-03-25 15:43:00 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

I just installed OpenMandriva Lx 2014 beta1 64 bit using [b]DVD[/b]: no change.

## Comment 4512
Date: 2014-03-26 13:32:54 +0000
From: @tpgxyz

(In reply to comment #5)
> I just installed OpenMandriva Lx 2014 beta1 64 bit using [b]DVD[/b]: no
> change.

How this correspond to this bug ? When people complains that we open CD/DVD....
https://issues.openmandriva.org/show_bug.cgi?id=579

## Comment 4531
Date: 2014-03-26 18:38:09 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

We (me and Davide) agree. I remember that installing alpha 1 using usb at the end of installation DVD reader opens. This is unuseful.

The right sequence could be: 

1) at the very end of installation stop all the activities and issue a message, something like "Remove your installation media (usb or dvd) and press Finish".
If it's not so difficult recognize the media used and chose between two different messages "Remove usb key used for installation and press Finish" or "Remove your installation dvd and press Finish"

2)opens dvd, better only if dvd is used to install. If it's easier for you open it anyway.
If it's not so difficult recognize the media used and open dvd reader only if dvd is used to install.

3) wait for Finish to be pressed. This give to the user enough time to remove installation media (either usb or disc)

4) only when Finish button is pressed reboot and complete installation process.

## Comment 4735
Date: 2014-04-01 13:17:05 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Strange behaviour in OpenMandriva 2014 beta1 64 bit:

1) installed using DVD no pause no disk eject

2) installed using USB memory no pause but disk drawer opens!

Please add a pause and a message that appears in both situations. If possible open drawer only if dvd is used.

## Comment 5356
Date: 2014-04-23 07:05:30 +0000
From: Denis Silakov &lt;<denis.silakov@rosalab.ru>&gt;

In ROSA, we are using a completely different approach to resolve the issue with installation media left in usb port or dvd recorder - in boot menu of the installation dvd, we use "Boot from local drive" item as the default. So there is no need to eject the media after installation, you can safely reboot your machine after installation is complete and without any actions from your side the newly installed system will be booted.

To be sure, this approach has a disadvantage that in order to start installation, you should manually select "Install" item in the boot menu. But a two-years experience of ROSA (and more longer experience of other systems such as SUSE) shows that this behavior is much more convinient for users. It is much more easier to press an arrow key and choose necessary menu item than to remember that you should eject your medium in time.

## Comment 5796
Date: 2014-05-03 08:05:56 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Still valid in OpenMandriva 2014 64 bit.

In some of the release I saw also automatic opening of dvd reader. Then the "complete solution " should not be so difficult to implement.  

If not possible to insert a pause at least copy Rosa solution that works much better than now.
(In reply to comment #9)
> It is much more easier to press an arrow key and choose necessary
> menu item than to remember that you should eject your medium in time.

## Comment 5827
Date: 2014-05-05 08:06:12 +0000
From: @tpgxyz

(In reply to comment #10)
> Still valid in OpenMandriva 2014 64 bit.
> 

This is fixed for CD/DVD medias. If user install OMV from optical drive, then this media is ejected.

## Comment 5887
Date: 2014-05-06 09:19:24 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #11)
> This is fixed for CD/DVD medias. If user install OMV from optical drive,
> then this media is ejected.
Installed twice OpenMadriva 2014 64 bit using dvd: dvd reader don't open.

## Comment 7306
Date: 2014-10-07 11:48:29 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

OpenMandriva 2014.1 64 bit installed in HD: still no dvd eject nor pause.

Is it so difficult to insert a pause just before reboot?

This bug is one year and half old !!!!!

## Comment 9799
Date: 2015-07-16 19:15:39 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Installed 2014.2 64 bit in HD using dvd: dvd tray open and close immediately because of reboot. No useful at all unless you're extremely fast.

Installed using USB memory and of course no pause at all.

Please read comment 7 and if too complicate read comment 9.

## Comment 11757
Date: 2016-02-07 13:19:54 +0000
From: @tapwag

Hello Giorgio,

thanks for reporting that issue. I changed the title of this bug report to indicate what issue the user is facing and more importantly that ROSA has a fix for this. 

I can't implement this change unfortunately as I lack the programming skills for this but I hope that someone will pick this up and also raised importance to "major" as this issue you are facing and have reported can be quite annoying. 

Thanks for this,

Maik

## Comment 11776
Date: 2016-02-07 18:14:12 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to Maik Wagner from comment #15)
> Hello Giorgio,
> 
> thanks for reporting that issue. I changed the title of this bug report to
> indicate what issue the user is facing and more importantly that ROSA has a
> fix for this. 
Thanks for looking at this very old bug.

> 
> I can't implement this change unfortunately as I lack the programming skills
> for this 
Neither me but for a developer should not be so difficult insert a pause!

>but I hope that someone will pick this up and also raised
> importance to "major" as this issue you are facing and have reported can be
> quite annoying. 
> 
> Thanks for this,
> 
> 
> Maik
This bug can't be solved for stable releases, including 2014.2.
But a very similar bug is present in cooker version just issued (bug #1446) hope that in OMLx 3 will be solved.

## Comment 12943
Date: 2016-04-20 16:30:46 +0000
From: @cris-b

*** Bug #1079 has been marked as a duplicate of this bug. ***

## Comment 12944
Date: 2016-04-20 16:31:28 +0000
From: @cris-b

we will fix for 3 not 2014.

## Comment 13329
Date: 2016-05-31 16:03:58 +0000
From: @tpgxyz

In Lx 3 we are using Calamares as installer. Please provide patches to calamares code base.

## Comment 13409
Date: 2016-06-03 18:13:09 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

In OMV Lx3 build 219 (issued on 25 may 2016) bug is still unsolved.

## Comment 19084
Date: 2019-02-06 20:43:12 +0000
From: @benbullard79

(In reply to Giorgio from comment #0)
> At the end of the first part of installation process installation media (usb
> or dvd) must be removed. This is not an easy task for beginner because you
> must remove media when pc is off and before reboot begin. 

This is incorrect. You can remove installation media at any time after reboot has begun. Everyone does it.
